pkg/util/archive/tar_xz.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package archive
+
+import (
+	"archive/tar"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/xi2/xz"
```

Comment:
Is there a reason to use this implementation vs something like https://github.com/ulikunitz/xz? This one doesn't seem to be actively maintained.

Reply:
It was the one used within `archiver` and labeled "fastxz" and only used for reading purposes. That is the only reason I chose it.

Reply:
The one you linked says this:
> The package is currently under development. There might be bugs and APIs are not considered stable.

Reply:
Ah, I see, thanks for explaining!